crates/dids/src/resolver.rs

```diff
@@ -0,0 +1,109 @@
+use crate::method::jwk::DidJwk;
+use crate::method::DidMethod;
+use serde::{Deserialize, Serialize};
+use ssi_dids::did_resolve::{
+    DocumentMetadata as DidDocumentMetadata, ResolutionMetadata as DidResolutionMetadata,
+};
+use ssi_dids::Document as DidDocument;
+
+pub struct DidResolver;
+
+impl DidResolver {
+    /// Resolves a DID URI, using the appropriate DID method, to a DID Document.
+    pub async fn resolve_uri(did_uri: &str) -> DidResolutionResult {
```

Comment:
So the idea here being, the DX is such that they can have a generalized `DidResolver.resolve_uri()` method to which they can pass in any URI and so long as it's one of our supported methods, and then get the resolution result back? 

This would be in contrast to a DX where the developer is already aware of their chosen DID Method, and they directly call the method resolution, for example `DidJwk.resolve_uri()`

I think that's a fine DX... okay and I can see we're offering [the same DX in `web5-js`](https://github.com/TBD54566975/web5-js/blob/main/packages/dids/src/did-resolver.ts#L63). So alright cool, just thinking out loud ✅

Reply:
Similar stuff is done in kt and golang. While this is fine, my main feedback is that this mechanism does not allow consumers to configure `DidResolver` to use consumer specified http_client libraries, nor which gateways to use for resolution.

Reply:
Gateways are really did-method specific, and I don't necessarily think that it's needed in this general "resolve this did_uri string for me" function. 

If a caller wants to resolve with a particular gateway, they probably already know which DidMethod they want to resolve with. They could directly call that resolver, with whatever options they want.

I'm a bit curious though on the http_client libraries. What did you have in mind there?

Reply:
@andresuribe87 What would you suggest here then? Should we modify this PR to add in options to resolve with a gateway? `did:jwk` doesn't take a gateway or do anything over http, so it would just be for future functionality right now. 

I don't mind the suggestion, but I don't know how we'd actionably do something with it here yet.